src/main/java/org/kiwiproject/registry/consul/server/ConsulRegistryService.java

```diff
@@ -0,0 +1,220 @@
+    }
+
+    @Override
+    public ServiceInstance updateStatus(ServiceInstance.Status newStatus) {
+        LOG.info("Ignoring status update as Consul handles it's own status through it's health check process");
+
+        if (isRegistered()) {
+            return registeredService.get().toBuilder().build();
+        }
+
+        return null;
+    }
+
+    @Override
+    public void unregister() {
+        checkState(isRegistered(), "Cannot unregister since registration was never called");
+
+        var serviceName = registeredService.get().getServiceName();
+        var instanceId = registeredService.get().getInstanceId();
+
+        LOG.info("Unregistering service {} with id {}", serviceName, instanceId);
+
+        var result = unregisterRetryer.tryGetObject(() -> {
+            consul.agentClient().deregister(registeredService.get().getInstanceId());
+            return instanceId;
+        });
+
+        Optionals.ifPresentOrElseThrow(result,
+                id -> {
+                    LOG.info("Service with name {} and id {} has been unregistered successfully", serviceName, instanceId);
+                    registeredService.set(null);
+                },
+                () -> {
+                    var msg = format("Error un-registering service {}, id {}", serviceName, instanceId);
+                    LOG.error(msg);
+                    return new RegistrationException(msg);
+                });
+    }
+
+    private boolean isRegistered() {
+        return nonNull(registeredService.get());
+    }
+
+    private Registration fromServiceInstance(ServiceInstance serviceInstance) {
+
+        var check = ImmutableRegCheck.builder()
+                .http(urlForPath(serviceInstance.getHostName(), serviceInstance.getPorts(), Port.PortType.ADMIN,
+                        serviceInstance.getPaths().getStatusPath()))
+                .interval(String.format("%ds", config.getCheckIntervalInSeconds()))
+                .deregisterCriticalServiceAfter(
+                        String.format("%dm", config.getDeregisterIntervalInMinutes()))
+                .build();
+
+        var address = adjustAddressIfNeeded(serviceInstance);
+
+        return ImmutableRegistration.builder()
+                .port(findFirstPortPreferSecure(serviceInstance.getPorts(), Port.PortType.APPLICATION).getNumber())
+                .check(check)
+                .id(UUIDs.randomUUIDString())
+                .name(serviceInstance.getServiceName())
+                .address(address)
+                .tags(List.of("service-type:default"))
+                .meta(Map.of(
+                        "version", serviceInstance.getVersion(),
+                        "commitRef", serviceInstance.getCommitRef(),
+                        "description", serviceInstance.getDescription(),
+                        "homePagePath", serviceInstance.getPaths().getHomePagePath(),
+                        "healthCheckPath", serviceInstance.getPaths().getHealthCheckPath(),
```

Comment:
What is the difference between healthcheckPath and statusCheckPath? Is the statusCheck essentially a ping and healthCheck the one that returns detailed check info?

Reply:
Correct, the defaults are `/ping` for status and `/healthcheck` for health check.  These are the DW defaults.